Actions/IncrementVersionNumber/IncrementVersionNumber.psm1

```diff
@@ -0,0 +1,171 @@
+﻿<#
+    .Synopsis
+        Changes a version setting value in a settings file.
+    .Parameter settingsFilePath
+        Path to a JSON file containing the settings.
+    .Parameter settingName
+        Name of the setting to change. The setting must be a version number.
+    .Parameter newValue
+        New value of the setting. Allowed values are: +1 (increment major version number), +0.1 (increment minor version number), or a version number in the format Major.Minor (e.g. 1.0 or 1.2
+#>
+function Set-VersionInSettingsFile {
+    param(
+        [Parameter(Mandatory = $true)]
+        [string] $settingsFilePath,
+        [Parameter(Mandatory = $true)]
+        [string] $settingName,
+        [Parameter(Mandatory = $true)]
+        [string] $newValue
+    )
+
+    #region Validate parameters
+    if (-not (Test-Path $settingsFilePath)) {
+        throw "Settings file ($settingsFilePath) not found."
+    }
+
+    Write-Host "Reading settings from $settingsFilePath"
+    try {
+        $settingFileContent = Get-Content $settingsFilePath -Encoding UTF8 | ConvertFrom-Json
+    }
+    catch {
+        throw "Settings file ($settingsFilePath) is malformed: $_"
+    }
+
+    if (-not ($settingFileContent.PSObject.Properties.Name -eq $settingName)) {
+        Write-Host "Setting $settingName not found in $settingsFilePath"
+        return
+    }
+
+    # Validate new version value
+    if ($newValue.StartsWith('+')) {
+        # Handle incremental version number
+
+        $allowedIncrementalVersionNumbers = @('+1', '+0.1')
+        if (-not $allowedIncrementalVersionNumbers.Contains($newValue)) {
+            throw "Incremental version number $newValue is not allowed. Allowed incremental version numbers are: $($allowedIncrementalVersionNumbers -join ', ')"
+        }
+    }
+    else {
+        # Handle absolute version number
+
+        $versionNumberFormat = '^\d+\.\d+$' # Major.Minor
+        if (-not ($newValue -match $versionNumberFormat)) {
+            throw "Version number $newValue is not in the correct format. The version number must be in the format Major.Minor (e.g. 1.0 or 1.2)"
+        }
+    }
+    #endregion
+
+    $oldValue = [System.Version] $settingFileContent.$settingName
+    $versions = @() # an array to hold the version numbers: major, minor, build, revision
```

Comment:
Minor: Confusing name versions - digits might be better

Reply:
Changed to `$versionNumbers`. Plus the comment, I hope it's clear enough.